publisher-sdk/src/test/java/com/criteo/publisher/logging/RemoteHandlerTest.kt

```diff
@@ -103,4 +108,17 @@ class RemoteHandlerTest {
 
     verify(sendingQueue).offer(logRecords)
   }
+
+  @Test
+  fun log_GivenConsentNotGiven_DoesNothing() {
+    whenever(consentData.isConsentGiven()).thenReturn(false)
+    val logMessage = LogMessage(message = null)
+    val logRecords = mock<RemoteLogRecords>()
+    whenever(remoteLogRecordsFactory.createLogRecords(logMessage)).thenReturn(logRecords)
```

Comment:
When there is no consent, we can even check that the factory is not used at all.
So maybe could you replace this stubbing by a `verify(factory, never()).createLogRecords(any())` ?